src/formatDate.js

```diff
@@ -8,7 +8,61 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[fromFormat.length - 1];
+  const toSeparator = toFormat[toFormat.length - 1];
+  const dateParts = date.split(`${fromSeparator}`);
+  let year = '';
+  let month = '';
+  let day = '';
+  const result = [];
+
+  for (let i = 0; i <= dateParts.length; i++) {
+    if (fromFormat[i] === 'YYYY') {
+      year = dateParts[i];
+    }
+
+    if (fromFormat[i] === 'YY') {
+      year = dateParts[i];
+    }
+
+    if (fromFormat[i] === 'MM') {
+      month = dateParts[i];
+    }
+
+    if (fromFormat[i] === 'DD') {
+      day = dateParts[i];
+    }
+  }
+
+  for (let i = 0; i < toFormat.length; i++) {
+    if (toFormat[i] === 'YY' && fromFormat.includes('YYYY')) {
+      result.push(year.slice(2));
+    }
+
+    if (toFormat[i] === 'YYYY' && fromFormat.includes('YY')) {
+      if (year < 30) {
```

Comment:
The comparison 'year < 30' may not work as expected because 'year' is a string. You should convert 'year' to a number before the comparison.